MCprep_addon/materials/default_materials.py

```diff
@@ -20,15 +20,18 @@
 import os
 
 import bpy
+from bpy.types import (
+  Context, Material
+)
 
 from .. import tracking
 from .. import util
 from . import sync
 
-from ..conf import env
+from ..conf import env, Engine
 
 
-def default_material_in_sync_library(default_material, context):
+def default_material_in_sync_library(default_material: list, context: Context) -> bool:
```

Comment:
Typically for typing of lists and dicts etc in python, you'd use the `from typing import List` or `Dict` or a more general structure called `Sequence`, I think `List` would be fine here rather than the built in. These are special cases which is why the typing library has the special imports.

Technically the _reason_ for doing this is outdated since python 3.8 [per here](https://stackoverflow.com/questions/39458193/using-list-tuple-etc-from-typing-vs-directly-referring-type-as-list-tuple-etc).

Right now we'll have build errors on Blender 2.80 which uses python 3.7 (different example, but shows that we _will_ need to make this fix everywhere, to uphold the standard of support back to 2.80)

```
  File "/Users/patrickcrawford/Library/Application Support/Blender/2.80/scripts/addons/MCprep_addon/materials/generate.py", line 251, in PrepOptions
    passes: dict[str]
TypeError: 'type' object is not subscriptable
```

In this example, we should use `from typing import Dict` and then use `Dict[str, Any]` or something like that.